SOF-7687: change propery schema title (for TS types)

k0stik opened the pull request:
New schemas:
[schema/properties_directory/structural/basis/boundary_conditions.json](https://github.com/Exabyte-io/esse/pull/359/files#diff-0b881ada9afb1746f175432ccba36f368398f47297e79bb5d536bf4e1d261db5)
[schema/properties_directory/structural/basis/predicted_properties.json](https://github.com/Exabyte-io/esse/pull/359/files#diff-c25e6ef4db7f476f6824943dfa892b69489cf20782d1090123b3df43b23b1dcc)
[schema/property/holder.json](https://github.com/Exabyte-io/esse/pull/359/files#diff-7a0e7a67b213f0de9fe16051362f9b9bced74d781386e232b202c3ce89c9c3db)
[schema/property/meta_holder.json](https://github.com/Exabyte-io/esse/pull/359/files#diff-cb09bb7170d6acf02a8553ffb4c3e41822f7d2e9885321945e8186fd13fd40ca)
[schema/property/proto_holder.json](https://github.com/Exabyte-io/esse/pull/359/files#diff-e2ed5f107db26ffd86780307beddfd171b459866b849c8e510aec0cc9fe7a33e)

timurbazhirov commented on schema/properties_directory/structural/basis/predicted_properties.json:
```diff
@@ -0,0 +1,28 @@
+        "name",
+        "values"
+    ]
+}
```
Should be outside `structural/basis` folder

timurbazhirov commented on schema/properties_directory/non-scalar/hubbard_v_nn.json:
```diff
@@ -1,12 +1,22 @@
     "allOf": [
         {
-            "$ref": "hubbard_v.json"
+            "$ref": "../../core/reusable/hubbard_parameters.json"
```
Let's add `reusable` inside properties_directory?

k0stik replied:
Done

timurbazhirov commented on schema/property/holder.json:
```diff
@@ -0,0 +1,189 @@
+                }
+            ]
+        },
+        "source": {
```
We could reuse source for meta and proto holders

k0stik replied:
I'm not sure we need this structure for Meta and Proto holders, as they contain an empty `source`:
```
"source" : {
        "info" : {

        },
        "type" : "exabyte"
    }
```
And holder has:
```
"source" : {
        "type" : "exabyte",
        "info" : {
            "jobId" : "HKPDxdJgGX5FrXLCy",
            "unitId" : "f23e15fa1304b005d942d64c-2ba70021fabbc28cbfb7cee0"
        }
    }
```

timurbazhirov commented on schema/properties_directory/non-scalar/is_relaxed.json:
```diff
@@ -0,0 +1,21 @@
+    "type": "object",
+    "allOf": [
+        {
+            "$ref": "../../material.json"
```
Do we need to store material here too?

k0stik replied:
Material exists in all `in_relaxed` properties in the production database

timurbazhirov commented on dist/js/example/property/base.json:
```diff
@@ -9,7 +9,8 @@
         "units": "kbar",
-        "value": -149.03
+        "value": -149.03,
+        "repetition": 0
```
Should not be here

k0stik replied:
Done

timurbazhirov commented on dist/js/example/property/base.json:
```diff
@@ -9,7 +9,8 @@
         "name": "pressure",
         "units": "kbar",
-        "value": -149.03
+        "value": -149.03,
```
the name of the file could be `holder.json` instead of base

k0stik replied:
Done

timurbazhirov commented on schema/workflow/base_flow.json:
```diff
@@ -17,26 +17,8 @@
-                {
-                    "type": "null"
-                }
-            ]
```
We should keep `compute`

k0stik replied:
Done

timurbazhirov commented on schema/core/abstract/2d_plot.json:
```diff
@@ -16,11 +16,6 @@
-        "legend": {
-            "description": "Legend of y Axis data series",
-            "minItems": 1,
-            "type": "array"
```
Don't we need legend anymore?

k0stik replied:
Based on production databases, we need `legend` only for the `density_of_states` property. Even more, the `legend` is required in the `density_of_states` property. It is already properly defined in `density_of_states.json`, file so I just removed it from here.

k0stik committed:
chore: comprehensive schema updates and improvements

- Add new coordinate condition schemas (cylinder, plane, sphere, triangular_prism)
- Add hubbard_parameters schema and jupyter_notebook_endpoint property
- Add final_structure and is_relaxed properties
- Add boundary_conditions and workflow properties
- Refactor property schemas: rename base.json to holder.json, add meta_holder and proto_holder
- Remove exabyteml and related ML schemas
- Update property schemas with required fields and improved validation
- Add repetition field to property schemas
- Improve band structure, phonon DOS, and other property schemas
- Update workflow schemas with better input/output handling
- Regenerate all Python models and TypeScript types
- Update package dependencies

k0stik committed:
chore: merge dev

timurbazhirov commented on src/py/mat3ra/esse/models/materials_category_components/entities/auxiliary/zero_dimensional/point_defect_site.py:
```diff
@@ -285,30 +285,38 @@ class VolumeSchema(BaseModel):
+class Name657(Enum):
+=======
 class Name573(Enum):
+>>>>>>> 8664c20e94ca9d52fce78f266358a60c7d88fb37
```
Somethins malfunctioned here - let's clean up the merge artifacts, pls.

k0stik replied:
Should all be fine now

timurbazhirov commented on src/py/mat3ra/esse/models/materials_category_components/entities/core/two_dimensional/vacuum.py:
```diff
@@ -369,7 +389,11 @@ class InChIRepresentationSchema(BaseModel):
+class Name804(Enum):
+=======
 class Name717(Enum):
+>>>>>>> 8664c20e94ca9d52fce78f266358a60c7d88fb37
```
Same as above

timurbazhirov commented on src/py/mat3ra/esse/models/materials_category_components/entities/reusable/three_dimensional/strained_uniform.py:
```diff
@@ -285,30 +285,38 @@ class VolumeSchema(BaseModel):
+class Name739(Enum):
+=======
 class Name653(Enum):
+>>>>>>> 8664c20e94ca9d52fce78f266358a60c7d88fb37
```
Same as above

timurbazhirov commented on src/py/mat3ra/esse/models/workflow/unit/__init__.py:
```diff
@@ -319,7 +323,11 @@ class ReduceUnitSchema(BaseModel):
+class Type91(Enum):
+=======
 class Type75(Enum):
+>>>>>>> 8664c20e94ca9d52fce78f266358a60c7d88fb37
```
Here, too

k0stik committed:
chore: new properties_directory/reusable dirrectory

k0stik committed:
chore: merge dev

k0stik committed:
chore: remove raw property

k0stik committed:
fix: core/reference/exabyte

k0stik committed:
fix: is-relaxed property

k0stik committed:
fix: is-relaxed property

k0stik committed:
update: required exabyteId

k0stik committed:
update: store only materialId in final_structure property

k0stik committed:
update: store only materialId in is_relaxed property

k0stik committed:
fix: schema id
